### PR TITLE
[BugFix] Vectorize TicTacToeEnv.win and honor rand_action(None)

### LINKED ISSUE #4342: [BUG] TicTacToeEnv.win is not batched and rand_action(None) crashes
## Describe the bug

`TicTacToeEnv` documents batched resets (`env.reset(TensorDict(batch_size=...))`), but `win()` uses Python `if tensor.sum() == 3`. That reduces over the whole batch, so one env completing a line can mark every env done, or a real win can be missed if the global sum is not 3.

`rand_action` also assumes a tensordict is passed and does `tensordict.get("mask")`. `EnvBase.rand_action(None)` is the public contract (same class of bug as #4225 on ChessEnv). `TicTacToeEnv().rand_action()` raises `AttributeError`.

## To Reproduce

```python
import torch
from tensordict import TensorDict
from torchrl.envs.custom.tictactoeenv import TicTacToeEnv

env = TicTacToeEnv()
env.rand_action()  # AttributeError

# Batched win: only env 0 has a completed row after the last action
```

## Expected behavior

- `win()` returns a per-env bool tensor (shape matching `turn` / `done`).
- `rand_action()` without a tensordict returns a TensorDict with `"action"`.
- With a reset tensordict, `rand_action` still honors `"mask"`.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the documentation (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by YeonwooSung**
## Description

`TicTacToeEnv.win` used Python `if tensor.sum() == 3`, which reduces over the whole batch. One env completing a line could mark every env done. `rand_action` also assumed a tensordict was passed, so `env.rand_action()` raised `AttributeError`.

`win()` now returns a per-env bool tensor. `rand_action(None)` builds a tensordict like ChessEnv (#4225) and still honors `"mask"` when present.

### Code example

A win in the first board must not mark the second board done. Sampling without an input TensorDict also works.

```python
import torch
from tensordict import TensorDict
from torchrl.envs import TicTacToeEnv

env = TicTacToeEnv()
assert "action" in env.rand_action()  # Previously raised AttributeError.
td = env.reset(TensorDict(batch_size=[2]))
board = td["board"]
board[0, 0, :2] = 1
board[0, 1, :2] = 0
board[1, 0, 0] = 1
board[1, 1, 0] = 0
td["mask"] = board.flatten(-2, -1) == -1
td["action"] = torch.tensor([2, 2])
next_td = env.step(td)["next"]
torch.testing.assert_close(next_td["done"], torch.tensor([[True], [False]]))
env.close()
```

## Motivation and Context

close #4342

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
Reviewed the batched win calculation, legal-action sampling, and single-player mixed-terminal update. All five TicTacToe tests passed locally. No actionable findings. GitHub CI has not run yet; the workflows are awaiting approval.
